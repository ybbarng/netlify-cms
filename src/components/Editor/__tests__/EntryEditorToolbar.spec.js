import React from 'react';
import { shallow } from 'enzyme';
import EditorToolbar from '../EditorInterface/EditorToolbar';

describe('EntryEditorToolbar', () => {
  it('should have the Save button disabled initally, and the Cancel button enabled', () => {
    const component = shallow(
      <EntryEditorToolbar
        onPersist={() => {}}
        onCancelEdit={() => {}}
        onDelete={() => {}}
      />
    );
    const tree = component.html();
    expect(tree).toMatchSnapshot();
  });

  it('should enable the Save button', () => {
    const component = shallow(
      <EntryEditorToolbar
        enableSave
        onPersist={() => {}}
        onCancelEdit={() => {}}
        onDelete={() => {}}
      />
    );
    const tree = component.html();
    expect(tree).toMatchSnapshot();
  });

  it('should disable and update label of Save button when persisting', () => {
    const component = shallow(
      <EntryEditorToolbar
        isPersisting
        onPersist={() => {}}
        onCancelEdit={() => {}}
        onDelete={() => {}}
      />
    );
    const tree = component.html();
    expect(tree).toMatchSnapshot();
  });
});
