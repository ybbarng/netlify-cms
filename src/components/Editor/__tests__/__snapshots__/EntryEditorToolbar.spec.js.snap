// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryEditorToolbar should disable and update label of Save button when persisting 1`] = `"<div><button disabled=\\"\\" class=\\"\\" type=\\"button\\" data-react-toolbox=\\"button\\">Saving...</button> <button class=\\"\\" type=\\"button\\" data-react-toolbox=\\"button\\">Cancel</button></div>"`;

exports[`EntryEditorToolbar should enable the Save button 1`] = `"<div><button class=\\"\\" type=\\"button\\" data-react-toolbox=\\"button\\">Save</button> <button class=\\"\\" type=\\"button\\" data-react-toolbox=\\"button\\">Cancel</button></div>"`;

exports[`EntryEditorToolbar should have the Save button disabled initally, and the Cancel button enabled 1`] = `"<div><button disabled=\\"\\" class=\\"\\" type=\\"button\\" data-react-toolbox=\\"button\\">Save</button> <button class=\\"\\" type=\\"button\\" data-react-toolbox=\\"button\\">Cancel</button></div>"`;
